include/zephyr/bluetooth/audio/cap.h

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_BLUETOOTH_AUDIO_CAP_H_
+#define ZEPHYR_INCLUDE_BLUETOOTH_AUDIO_CAP_H_
+
+/**
+ * @brief Common Audio Profile (CAP)
+ *
+ * @defgroup bt_cap Common Audio Profile (CAP)
+ *
+ * @ingroup bluetooth
+ * @{
+ *
+ * [Experimental] Users should note that the APIs can change
+ * as a part of ongoing development.
+ */
+
+#include <zephyr/types.h>
+#include <bluetooth/audio/csis.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief Register the Common Audio Service
+ *
+ * This will register and enable the service and make it discoverable by
+ * clients. This will also register a Coordinated Set Identification
+ * Service instance.
+ *
+ * This shall only be done as a server, and requires
+ * @kconfig{BT_CAP_ACCEPTOR_SET_MEMBER}. If @kconfig{BT_CAP_ACCEPTOR_SET_MEMBER}
+ * is not enabled, the Common Audio Service will by statically registered.
+ *
+ * @param[in]  param Coordinated Set Identification Service register parameters.
+ * @param[out] csis  Pointer to the registered Coordinated Set Identification
+ *                   Service.
+ *
+ * @return 0 if success, errno on failure.
+ */
+int bt_cap_acceptor_register(const struct bt_csis_register_param *param,
+			     struct bt_csis **csis);
```

Comment:
I am not sure this function does what it's expected to do. It's named `bt_cap_acceptor_register` which implies that it will register CAP profile role, but it does it's registration on GATT CAS instance.
I guess that it's insufficient. Is it really needed to have a such function? Why the CAS could not be registered if some Kconfig option e.g. `BT_AUDIO_ROLE_ACCEPTOR` is set?

The CAP specification says:
```
An Acceptor:
• Can transmit and receive unicast Audio Streams
• Can expose Audio Stream Endpoints (ASEs)
• Can receive broadcast Audio Streams
• Signals availability and support for different use cases as described by Context Type values
• Receives context on the intended use case of unicast or broadcast Audio Streams as described by
Context Type values
• Receives information on the association of unicast or broadcast Audio Streams with Content Control
services through CCID values
• Can scan for broadcast Audio Streams
• Can delegate scanning for broadcast Audio Streams to a Commander
• Can request and receive a Broadcast_Code
• Can receive requests from a Commander to start or stop reception of broadcast Audio Streams
• Can receive requests from a Commander to mute and set the volume of audio that the Acceptor is
rendering
• Can receive requests from a Commander to mute and set the signal level of its microphone
• Can host CCP and MCP clients
```

Reply:
This response is also for https://github.com/zephyrproject-rtos/zephyr/pull/44522#discussion_r864527215:

Basically I would have liked to do static allocation only. This is still possible, but the CSIS is dynamically registered and require a set of parameters to be registered. I ended up with this dynamic allocation for CAS with CSIS, such that the upper layer can provide those parameters to the CAS CSIS instance. 
It would be possible to add Kconfig options for the parameters for the CAS CSIS instance, and that way make it statically registered, even when the CAP Acceptor is a set member. 

Regarding the name (if we keep it as dynamic), we can rename it to `bt_cap_csis_register` or `bt_cap_acceptor_register_csis` (to keep the profile role as the function prefix) if that is better (if we add Kconfigs for the CSIS parameters and always statically register CAS, then this function goes away)